packages/config-loader/src/loader.ts

```diff
@@ -176,7 +179,7 @@ export async function loadConfig(
     return { fileConfigs, loadedPaths };
   };
 
-  const loadRemoteConfigFiles = async () => {
+  const loadRemoteConfigFiles = async (): Promise<AppConfig[]> => {
```

Comment:
Fine with skipping these changes to `config-loader`? Otherwise we'll need to split this out into a separate changeset as it's just an internal type tweak.